src/ert/_c_wrappers/enkf/enkf_main.py

```diff
@@ -378,25 +386,90 @@ def initRun(self, run_context: "RunContext", parameters: List[str] = None):
-                        or current_status == STATE_LOAD_FAILURE,
-                    ):
-                        enkf_state.state_initialize(
-                            rng,
-                            enkf_node,
-                            run_context.sim_fs,
-                            realization_nr,
+
+                    if (
+                        not enkf_node.has_data(
+                            run_context.sim_fs, NodeId(0, realization_nr)
                         )
+                        or current_status == STATE_LOAD_FAILURE
+                    ):
+                        rng = self.realizations[realization_nr]
+                        impl_type = enkf_node.getImplType()
+
+                        if impl_type == ErtImplType.GEN_KW:
+                            file = self._alloc_initfile(config_node, realization_nr)
+
+                            if self._initialize_gen_kw(
+                                enkf_node.asGenKw(),
+                                rng,
+                                file,
+                            ):
+                                vals = np.array(
+                                    [i[1] for i in enkf_node.asGenKw().items()]
+                                )
+
+                                s = io.BytesIO()
+                                # The first element is time_t (64 bit integer), but it is
+                                # not used so we write 0 instead - for padding purposes.
+                                s.write(struct.pack("Qi", 0, int(ErtImplType.GEN_KW)))
+                                s.write(vals.tobytes())
+
+                                enkf_fs.write_parameter(
+                                    run_context.sim_fs,
+                                    config_node.getKey(),
+                                    realization_nr,
+                                    s.getvalue(),
+                                )
+
+                        else:
+                            enkf_state.state_initialize(
+                                self.realizations[realization_nr],
+                                enkf_node,
+                                run_context.sim_fs,
+                                realization_nr,
+                            )
+
                 if state_map[realization_nr] in [STATE_UNDEFINED, STATE_LOAD_FAILURE]:
                     state_map[realization_nr] = RealizationStateEnum.STATE_INITIALIZED
         run_context.sim_fs.sync()
 
+    def _alloc_initfile(self, config_node: EnkfConfigNode, iens: int) -> Optional[Path]:
+        file_format = config_node.get_init_file_fmt()
+
+        if not file_format:
+            return None
+
+        return Path(
+            file_format % iens if "%d" in file_format else file_format
+        ).absolute()
+
+    def _initialize_gen_kw(
+        self, gen_kw: GenKw, rng: RandomNumberGenerator, file: Optional[Path]
+    ) -> bool:
+        if file is not None and file.exists():
```

Comment:
Is this the same order as before? Might be that it is, but is potentially some confusing behavior, perhaps update the docs? Also, is there a test for the two column GEN_KW?